share/completions/dnf.fish

```diff
@@ -2,4 +2,78 @@
 # Completions for the dnf command
 #
 
-complete -c dnf -w yum
+# All dnf commands
+
+# Test if completing using package names is appropriate
+function __fish_dnf_package_installed
+	for i in (commandline -poc)
+		if contains -- $i update upgrade remove erase reinstall
+			return 0
+		end
+	end
+	return 1
+end
+
+
+function __fish_dnf_package_notinstalled
+	for i in (commandline -poc)
+		if contains -- $i install
+			return 0
+		end
+	end
+	return 1
+end
+
+# Installed packages
+function __dnf_installed
+	sqlite3 /var/cache/dnf/packages.db "select pkg from installed WHERE pkg LIKE \"$cur%\"" 2>/dev/null
+	return
+end
+
+# Available packages
+function __dnf_available
+	sqlite3 /var/cache/dnf/packages.db "select available.pkg from available left join installed on available.pkg = installed.pkg where installed.pkg IS NULL and available.pkg LIKE \"$cur%\"" 2>/dev/null
+	return
+end
+
+
+complete -c dnf -n '__fish_use_subcommand' -xa install --description "Install the latest version of a package"
```

Comment:
I think installing the latest version is so common across package managers (and also just the sensible thing to do) that it doesn't need to be mentioned. That cuts down on the length of the description, which potentially allows us to use more columns, which makes navigation faster.